### PR TITLE
[@astrojs/image] adding caching support for SSG builds

### PR DESCRIPTION
**Pull request by tony-sull**
## Changes

Adds caching support for transformed images :tada:

## Testing

Tests added to verify images are cached during SSG builds

## Docs

integration README updated with details on how to configure or disable caching

 /cc @withastro/maintainers-docs

## Tested in Netlify with an astro.build PR build

> First run (no images cached)
Screen Shot 2022-09-29 at 11 08 53 AM

> Second run (1 image file updated, the rest cached)
Screen Shot 2022-09-29 at 11 09 21 AM

### CODE REVIEW

File: packages/integrations/image/src/build/ssg.ts [hunk 1]
```diff
@@ -67,6 +118,9 @@ export async function ssgBuild({ loader, staticImages, config, outDir, logLevel
 		let inputFile: string | undefined = undefined;
 		let inputBuffer: Buffer | undefined = undefined;
 
+		// tracks the cache duration for the original source image
+		let expires: number = 0;
```
Comment (matthewp) [inline packages/integrations/image/src/build/ssg.ts:122]: eslint doesn't like you typing this :/

Comment (tony-sull) [reply packages/integrations/image/src/build/ssg.ts]: Yep, kind of an aggressive tslint rule there 🤷

**Comment by natemoo-re** [comment]
Just wanted to check that you saw https://github.com/withastro/astro/pull/4765 ? I don't think we need to leap to a final, built-in abstraction just yet, but it'd be great if this work for `@astrojs/image` was agnostic enough that it could lead to a generic caching mechanism for other integrations down the road.

**Comment by tony-sull** [comment]
> Just wanted to check that you saw https://github.com/withastro/astro/pull/4765 ? I don't think we need to leap to a final, built-in abstraction just yet, but it'd be great if this work for @astrojs/image was agnostic enough that it could lead to a generic caching mechanism for other integrations down the road.

Agreed!  I'm hoping this can turn into a general caching solution for Astro.  I didn't go down the path of a full API for core here, but there really isn't anything image-specific in this implementation :+1:

**Review by matthewp — APPROVED** [review]

**Pull request merged**